apps/uriparser.cpp

```diff
@@ -115,17 +116,17 @@ void UriParser::Parse(const string& strUrl, DefaultExpect exp)
     if (idx != string::npos)
     {
         m_proto = m_host.substr(0, idx);
+        transform(m_proto.begin(), m_proto.end(), m_proto.begin(), [](char c){ return std::toupper(c); });
```

Comment:
Micosoft VC reports an error here; looxlike not every compiler puts the standard C functions into the `std` namespace.

Also `std::` is superfluous here, as you are also using `transform` call without `std::` and it's not being complained about.

The use of `std::` prefix (and disallowed `using namespace std`) is only in header files, as such a declaration would spread to every `*.cpp` file as well as all next `*.h` files included. Here is `using namespace std` in the beginning.

Reply:
I checked this with removed `std::` prefix and it compiles without error on both Linux and Windows.

BTW. didn't notice before - why do you use `toupper` here? Please use `tolower`, there's lots of code already that checks the `scheme()` result against a lowercase scheme. Note that the enumeration for schemes is only for quick check for the most known schemes, but this UriParser is also predicted to be used for cases when the scheme is "generic" and it can only be recognized through the string comparison.

Reply:
`toupper` crept in while I was trying to investigate what was happening with namespacing etc. Fixed now.